apis/storage/v1alpha3/bucketpolicy_types.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// S3BucketPolicyParameters define the desired state of an AWS S3BucketPolicy.
+type S3BucketPolicyParameters struct {
+	// This is the current IAM policy version
+	PolicyVersion string `json:"Version"`
+
+	// This is the policy's optional identifier
+	PolicyID string `json:"Id,omitempty"`
+
+	// This is the list of statement this policy applies
+	PolicyStatement []S3BucketPolicyStatement `json:"Statement"`
+}
+
+//Serialize is the custom marshaller for the S3BucketPolicyParameters
+func (p *S3BucketPolicyParameters) Serialize() (interface{}, error) {
+	m := make(map[string]interface{})
+	m["Version"] = p.PolicyVersion
+	if p.PolicyID != "" {
+		m["Id"] = p.PolicyID
+	}
+	slc := make([]interface{}, len(p.PolicyStatement))
+	for i, v := range p.PolicyStatement {
+		msg, err := v.Serialize()
+		if err != nil {
+			return nil, err
+		}
+		slc[i] = msg
+	}
+	m["Statement"] = slc
+	//dat, err := json.Marshal(m)
+	//if err != nil {
+	//	return nil, err
+	//}
+	//str := string(dat)
+	return m, nil
+}
+
+// S3BucketPolicyStatement defines an individual statement within the
+// S3BucketPolicyBody
+type S3BucketPolicyStatement struct {
+	// Optional identifier for this statement, must be unique within the
+	// policy if provided.
+	StatementID string `json:"Sid,omitempty"`
```

Comment:
```suggestion
	StatementID string `json:"sid,omitempty"`
```

JSON tags should start with lowercase